include/nlohmann/json.hpp

```diff
@@ -1209,25 +1210,25 @@ class basic_json // NOLINT(cppcoreguidelines-special-member-functions,hicpp-spec
             default:
                 break;
         }
-#if DIAGNOSTIC_POSITIONS
+
+#if JSON_DIAGNOSTIC_POSITIONS
         start_position = other.start_position;
         end_position = other.end_position;
 #endif
 
         set_parents();
         assert_invariant();
-
     }
 
     /// @brief move constructor
     /// @sa https://json.nlohmann.me/api/basic_json/basic_json/
     basic_json(basic_json&& other) noexcept
         : json_base_class_t(std::forward<json_base_class_t>(other)),
-#if DIAGNOSTIC_POSITIONS
-          start_position(other.start_position),
-          end_position(other.end_position),
-#endif
           m_data(std::move(other.m_data))
+#if JSON_DIAGNOSTIC_POSITIONS
+        , start_position(other.start_position),
```

Comment:
Compiler doesn’t like it being before m_data, any suggestions?

Reply:
It would require reordering the members in the class definition, not just here in this constructor.  I think it's probably better to just put the commas first and align them with `m_data`.

Reply:
Yes, it's ugly, but changing the order everywhere else is even uglier.